ambry-named-mysql/src/integration-test/java/com/github/ambry/named/MySqlNamedBlobDbIntegrationTest.java

```diff
@@ -248,6 +248,7 @@ public void testCleanupBlobsPipeline() throws Exception {
 
     int staleCount = 10;
     int needCleanupCount = 0;
+    List<NamedBlobRecord> staleRecords = new ArrayList<>();
```

Comment:
is there a new test case? Can we add a test case where the blob id is shared by two versions? and see if the query solve this issue?

Reply:
Yes, that's tested in `testCleanupBlobsPipeline`, which actually failed after using the new query in `MySqlNamedBlobDb`. I need to make the change to make it pass, and as you can see it only delete the stale rows that do not share valid blob_id with latest record. I have to use a blobId2 to create a different blobId case which can be cleaned up:
```
      String blobId2 = getBlobId(account, container);
      NamedBlobRecord record2 =
          new NamedBlobRecord(account.getName(), container.getName(), blobName, blobId2, expirationTime);
      namedBlobDb.put(record2, NamedBlobState.READY, true).get();
```